lib/instrumentation/modules/amqplib.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+var shimmer = require('../shimmer')
+var { getDBDestination } = require('../context')
+
+var URL = require('url-parse')
+
+module.exports = function (amqplib, agent, { version, enabled }) {
+  agent.logger.debug('shimming amqplib.connect')
+  shimmer.wrap(amqplib, 'connect', wrapConnect)
+
+  return amqplib
+
+  function patchChannelModel (channelModel, host, port) {
+    agent.logger.debug('shimming amqplib.ChannelModel.connection.sendMessage')
+    shimmer.wrap(channelModel.connection, 'sendMessage', function (orig) {
+      return function (channel, method, fields, properties, props, content) {
+        var span = enabled && agent.startSpan(null, 'messaging', 'amqp', 'publish')
```

Comment:
Why starting an span and not a transaction? I'm not pretty sure but I think that Express, for instance, starts with a transaction because if not, you should create a transaction before express handles the route request. I'm right?